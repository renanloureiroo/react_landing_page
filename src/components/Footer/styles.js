import styled, { css } from 'styled-components';

import { TextComponent } from '../TextComponent/index';
export const Container = styled.div`
  ${({ theme }) => css`
    & a {
      color: inherit;
      text-decoration: none;
      font-size: ${theme.sizes.small};
    }
    text-align: center;
  `}
`;

export const TextFooter = styled(TextComponent).attrs('div')``;
