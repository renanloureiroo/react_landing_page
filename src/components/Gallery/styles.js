import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${() => css``}
`;
export const TextContainer = styled.div`
  ${({ theme }) => css`
    margin-bottom: ${theme.spacing.huge};
  `}
`;
export const GridContainer = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(3, 1fr);
    justify-content: center;
    gap: ${theme.spacing.medium};

    & div {
      width: 100%;
      height: 100%;
      margin: 0 auto;
      overflow: hidden;
    }

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
    }
  `}
`;
export const Grid = styled.img`
  transition: all 300ms ease-in-out;
  ${() => css`
    width: 100%;
    height: 100%;

    &:hover {
      transform: scale(1.2) rotate(7deg);
    }
  `}
`;
