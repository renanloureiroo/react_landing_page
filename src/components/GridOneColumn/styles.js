import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${() => css`
    max-width: 50%;
    margin: 0 auto;
    text-align: center;
  `}
`;
