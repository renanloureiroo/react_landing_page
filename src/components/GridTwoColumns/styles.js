import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: 1fr 1.5fr;
    gap: ${theme.sizes.small};

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
    } ;
  `}
`;

export const TextContainer = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-flow: column nowrap;
    justify-content: center;

    & h1 {
      margin-bottom: ${theme.sizes.small};
    }
  `}
`;
export const ImgContainer = styled.div`
  ${() => css`
    width: 100%;
    height: 100%;
    margin: 0 auto;

    & img {
      width: 100%;
      height: 100%;
    }
  `}
`;
