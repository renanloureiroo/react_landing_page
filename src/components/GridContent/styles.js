import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    display: grid;
    margin: 0 auto;
    max-width: 58rem;
    grid-template-columns: 1fr;
    text-align: center;
    justify-content: center;
    ${Title} {
      margin-bottom: ${theme.spacing.large};
    }
  `}
`;

export const Html = styled.div`
  ${({ theme }) => css`
  & p{
    margin-bottom: ${theme.spacing.large}};
  }
  `}
`;
