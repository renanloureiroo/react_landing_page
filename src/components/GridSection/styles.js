import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-rows: 1fr 2fr;
    gap: ${theme.spacing.large};

    ${Title} {
      margin-bottom: ${theme.spacing.small};
    }

    @media ${theme.media.lteMedium} {
      display: block;
    }
  `}
`;
export const TextContainer = styled.div`
  ${({ theme }) => css`
    margin-bottom: ${theme.spacing.large};
  `}
`;

export const GridContainer = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(3, 1fr);
    gap: ${theme.spacing.medium};
    counter-reset: grid-counter;

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
    }
  `}
`;
export const Grid = styled.div`
  ${({ theme }) => css`
    max-width: 36rem;
    ${Title} {
      position: relative;
      left: 5rem;
    }

    ${Title}::before {
      counter-increment: grid-counter;
      content: counter(grid-counter);
      position: absolute;
      font-size: ${theme.spacing.xhuge};
      font-weight: bold;
      transform: rotate(10deg);
      left: -5rem;
      top: -2rem;
    }
  `}
`;
