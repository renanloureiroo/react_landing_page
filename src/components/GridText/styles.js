import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-rows: 1fr 2fr;
    gap: ${theme.spacing.large};

    ${Title} {
      margin-bottom: ${theme.spacing.small};
    }

    @media ${theme.media.lteMedium} {
      display: block;
    }
  `}
`;
export const TextContainer = styled.div`
  ${({ theme }) => css`
    margin-bottom: ${theme.spacing.large};
  `}
`;

export const GridContainer = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(28rem, 1fr));
    gap: ${theme.spacing.large};
    counter-reset: grid-counter;

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
    }
  `}
`;
export const Grid = styled.div`
  ${({ theme }) => css`
    margin-bottom: ${theme.spacing.medium};
    ${Title} {
      position: relative;
      left: 5rem;
    }

    ${Title}::before {
      counter-increment: grid-counter;
      content: counter(grid-counter);
      position: absolute;
      font-size: ${theme.spacing.xhuge};
      font-weight: bold;
      transform: rotate(10deg);
      left: -5rem;
      top: -2rem;
    }
  `}
`;
