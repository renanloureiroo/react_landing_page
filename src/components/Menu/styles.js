import styled, { css } from 'styled-components';
import { Container as SectionContainer } from '../SectionContainer/styles';
import { Title as Heading } from '../Heading/styles';

// eslint-disable-next-line no-unused-vars
const menuVisible = () => css`
  visibility: visible;
  opacity: 1;
`;

export const Container = styled.div`
  ${({ theme, visible }) => css`
    position: fixed;
    z-index: 5;
    top: 0;
    right: 0;
    left: 0;
    border-bottom: 0.1rem solid ${theme.colors.mediumGray};
    transition: all 300ms ease-in-out;

    & ${SectionContainer} {
      padding-top: 0;
      padding-bottom: 0;
    }

    @media ${theme.media.lteMedium} {
      min-height: 100vh;
      margin-bottom: 0;
      visibility: hidden;
      opacity: 0;
      ${visible && menuVisible()}

      & ${SectionContainer} {
        display: grid;
        grid-template-columns: 1fr;
        grid-template-rows: 1fr;
        align-items: center;
        height: 100vh;
        overflow-y: auto;
      }
      & ${Heading} {
        padding-bottom: ${theme.spacing.large};
        display: flex;
        justify-content: center;
      }
    } ;
  `}
`;
export const MenuContainer = styled.div`
  ${({ theme }) => css`
    display: flex;
    justify-content: space-between;
    align-items: center;

    @media ${theme.media.lteMedium} {
      display: block;
      text-align: center;
    } ;
  `}
`;
export const Button = styled.button`
  ${({ theme }) => css`
    cursor: pointer;
    display: none;
    position: fixed;
    z-index: 6;
    right: 2rem;
    top: 2rem;
    width: 4rem;
    height: 4rem;
    background: ${theme.colors.primaryColor};
    color: ${theme.colors.white};
    border: none;
    border-radius: 10%;

    & svg {
      width: 3.5rem;
      height: 3.5rem;
    }

    @media${theme.media.lteMedium} {
      display: block;
    }
  `}
`;
