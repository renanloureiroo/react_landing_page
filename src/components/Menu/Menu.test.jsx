import { fireEvent, screen } from '@testing-library/react';
import { renderTheme } from '../../styles/renderTheme';

import { Menu } from '.';
import Links from '../NavLinks/mock';
import { theme } from '../../styles/theme';

const logoData = {
  text: 'Logo',
  srcImg: 'assets/images/javascript.svg',
  link: '#target',
};

describe('<Menu />', () => {
  it('should render Logo and Main Menu Nav', () => {
    const { container } = renderTheme(
      <Menu Links={Links} logoData={logoData} />,
    );
    expect(screen.getByRole('heading', { name: 'Logo' })).toBeInTheDocument();
    expect(
      screen.getByRole('navigation', { name: 'Main menu' }),
    ).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render menu mobile and button for open and close the menu', () => {
    renderTheme(<Menu Links={Links} logoData={logoData} />);

    const button = screen.getByLabelText('Open/Close menu');
    const menuContainer = button.nextSibling;

    expect(button).toHaveStyleRule('display', 'none');
    expect(button).toHaveStyleRule('display', 'block', {
      media: theme.media.lteMedium,
    });
    expect(menuContainer).toHaveStyleRule('opacity', '0', {
      media: theme.media.lteMedium,
    });
    expect(screen.getByLabelText('Open menu')).toBeInTheDocument();

    fireEvent.click(button);
    expect(menuContainer).toHaveStyleRule('opacity', '1', {
      media: theme.media.lteMedium,
    });
    expect(screen.getByLabelText('Close menu')).toBeInTheDocument();

    fireEvent.click(menuContainer);
    expect(menuContainer).toHaveStyleRule('opacity', '0', {
      media: theme.media.lteMedium,
    });
    expect(screen.getByLabelText('Open menu')).toBeInTheDocument();
  });

  it('should not render links', () => {
    const { container } = renderTheme(<Menu logoData={logoData} />);
    expect(
      screen.queryByRole('navigation', { name: 'Main menu' }).firstChild,
    ).not.toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
