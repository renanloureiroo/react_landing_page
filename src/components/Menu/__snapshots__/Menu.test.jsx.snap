// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> should not render links 1`] = `
.c0 {
  cursor: pointer;
  display: none;
  position: fixed;
  z-index: 6;
  right: 2rem;
  top: 2rem;
  width: 4rem;
  height: 4rem;
  background: #0A1128;
  color: #fff;
  border: none;
  border-radius: 10%;
}

.c0 svg {
  width: 3.5rem;
  height: 3.5rem;
}

.c1 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

@media (max-width:768px) {

}

@media (max-width:768px) {

}

@media (max-width:768px) {

}

@media (max-width:768px) {
  .c0 {
    display: block;
  }
}

@media (max-width:768px) {

}

<button
  aria-label="Open/Close menu"
  class="c0"
>
  <svg
    aria-label="Open menu"
    class="c1"
    fill="currentColor"
    focusable="false"
    role="img"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title>
      Open menu
    </title>
    <path
      d="M0 0h24v24H0V0z"
      fill="none"
    />
    <path
      d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
    />
  </svg>
</button>
`;

exports[`<Menu /> should render Logo and Main Menu Nav 1`] = `
.c0 {
  cursor: pointer;
  display: none;
  position: fixed;
  z-index: 6;
  right: 2rem;
  top: 2rem;
  width: 4rem;
  height: 4rem;
  background: #0A1128;
  color: #fff;
  border: none;
  border-radius: 10%;
}

.c0 svg {
  width: 3.5rem;
  height: 3.5rem;
}

.c1 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

@media (max-width:768px) {

}

@media (max-width:768px) {

}

@media (max-width:768px) {

}

@media (max-width:768px) {
  .c0 {
    display: block;
  }
}

@media (max-width:768px) {

}

<button
  aria-label="Open/Close menu"
  class="c0"
>
  <svg
    aria-label="Open menu"
    class="c1"
    fill="currentColor"
    focusable="false"
    role="img"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title>
      Open menu
    </title>
    <path
      d="M0 0h24v24H0V0z"
      fill="none"
    />
    <path
      d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
    />
  </svg>
</button>
`;
